Lombiq.Tests.UI/Models/UserLoginParameters.cs

```diff
@@ -0,0 +1,56 @@
+using Lombiq.Tests.UI.Extensions;
+using Lombiq.Tests.UI.Helpers;
+using Lombiq.Tests.UI.Services;
+using OpenQA.Selenium;
+using System;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Lombiq.Tests.UI.Models;
+
+public record UserLoginParameters(
+    string UserName,
+    string Password,
+    string LoginButtonText = UserLoginParameters.DefaultLoginButtonText)
+{
+    public const string DefaultLoginButtonText = "Log in";
+
+    public UserLoginParameters(UserRegistrationParameters parameters)
+        : this(parameters.UserName, parameters.Password, parameters.LoginButtonText)
+    {
+    }
+
+    public async Task LogInAsync(UITestContext context, bool navigate = true)
```

Comment:
Having such an operation in a model class doesn't look appropriate. Add a `UITestContext` extension instead that expects this class as the input. Same for `UserRegistrationParameters.RegisterAsync` and `OrchardCoreSetupParameters.SetupOrchardCoreAsync`.

Reply:
Why is it inappropriate? It's a UI test model, using its own data in UI test seems like valid OOP to me. Makes the code more discoverable too, as opposed to having to find the logic in one of the million `*UITestContextExtensions` classes.